pkg/controller/workload/kubernetes/resource/resource.go

```diff
@@ -216,7 +216,11 @@ func (c *remoteCluster) delete(ctx context.Context, ar *v1alpha1.KubernetesAppli
 	ensureNamespace(template)
 	setRemoteController(ar, template)
 
-	if err := c.unstructured.delete(ctx, ar.Spec.Template.Raw); err != nil {
+	raw, err := json.Marshal(template)
+	if err != nil {
+		return v1alpha1.KubernetesApplicationResourceStateFailed, errors.Wrap(err, errUnmarshalTemplate)
+	}
+	if err := c.unstructured.delete(ctx, raw); err != nil {
```

Comment:
I'm not a huge fan of needing to marshal to unstructured, set some metadata, then marshal back to YAML, which is again unmarshalled to unstructured in the unstructuredClient. Would it make more sense for the unstructuredClient to accept and operate on `*unstructured.Unstructured` and unmarshal once in the `remoteCluster`?

Reply:
@negz I agree that it is a bit cumbersome. However, it feels like it makes the point of using `raw.Extension` a little less impactful if we are still passing around `*unstructured.Unstructured`. That being said, I think the point of passing around `[]bytes` is to guard against direct modification of the object, and that may not be as important in this situation where there are not external consumers of these methods. What do you think? Am I off base in how I am thinking about the value of using `raw.Extension`?

Reply:
@hasheddan I believe the point of using `raw.Extension` is mostly to avoid embedding a real, functional `runtime.Object` inside another `runtime.Object`. When we embed a real `Object` we must always remember to deepcopy it to avoid any changes we make being propagated back to our "template".

I believe we can fulfil this goal by simply unmarshaling a `raw.Extension` early on in the reconcile, close to where we parse the KAR struct. We'll then have an *unstructured.Unstructured` we can plumb down into our various helpers, clients, etc without those methods having to worry about deepcopying before modifying.

Apologies that I didn't catch this in the previous PR!

Reply:
@negz no worries! Could you give this another look? I am going to run manual testing again to confirm operations :+1: